### PR TITLE
[containerapp] az containerapp add-on create: Support for weaviate add-on

### PR DESCRIPTION
**Pull request by bgashirabake**
---

****Changes:**
-support new add-on weaviate
-deprecate service command group by next cli core version release**

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

**Comment by bgashirabake** [comment]
**NOTE:
- i got multiple flake8 warnings and errors that previously would pass the CI workflow but i didn't know if should run autopep8 on the extension to fix it. please advise if i should run it to fix the linting errors on all azext_containerapp files. (running `azdev style containerapp` returns no error but running `flake8 .` anywhere does.)
- the failing integration tests are unrelated to my change.**

File: src/containerapp/azext_containerapp/commands.py [hunk 1]
```diff
@@ -50,52 +50,56 @@ def load_command_table(self, _):
     with self.command_group('containerapp env dapr-component') as g:
         g.custom_command('init', 'init_dapr_components', is_preview=True)
 
-    with self.command_group('containerapp service', deprecate_info=self.deprecate(redirect='containerapp add-on', hide=True), is_preview=True) as g:
+    with self.command_group('containerapp service', deprecate_info=self.deprecate(redirect='containerapp add-on', expiration='2.56.1', hide=True), is_preview=True) as g:
```
Comment (Greedygre) [inline src/containerapp/azext_containerapp/commands.py:53]: Can you show me a picture about what it will looks like after added `expiration='2.56.1'`?

Comment (bgashirabake) [reply src/containerapp/azext_containerapp/commands.py:53]: image

Comment (Greedygre) [reply src/containerapp/azext_containerapp/commands.py:53]: @bgashirabake The command group `containerapp service` only exist in azure-cli-extension, the expiration version should be extension version.

**Review by Greedygre — APPROVED** [review]

**Comment by Greedygre** [comment]
Hi @zhoxing-ms @wangzelin007 

Could you help to review this PR?

The azdev style check got lots of flake8 warnings, I will handle them in other PR. Thanks.

File: src/containerapp/azext_containerapp/commands.py (hunk 1 above)
Comment (zhoxing-ms) [reply src/containerapp/azext_containerapp/commands.py:53]: Please note that after adding `expiration='2.56.1'`, if the customer is using a version greater than `2.56.1` , these commands will not work and cause breaking change to the customer's automation scripts

Comment (bgashirabake) [reply src/containerapp/azext_containerapp/commands.py:53]: They should use add-on instead

Comment (Greedygre) [reply src/containerapp/azext_containerapp/commands.py:53]: Hi @bgashirabake 

Please note the breaking change on the `history.rst ` about if the customer is using the Azure CLI version greater than 2.56.1 , these commands will not work and cause breaking change.
For example:
```
* [Breaking Change]  'az containerapp service': deprecate from Azure CLI version 2.56.1
* 'az containerapp add-on' : support for add-on weaviate create and delete commands
```

File: src/containerapp/HISTORY.rst [hunk 2]
```diff
@@ -4,6 +4,7 @@ Release History
 ===============
 upcoming
 ++++++
+* 'az containerapp add-on' : support for add-on weaviate create and delete commands
```
Comment (Greedygre) [inline src/containerapp/HISTORY.rst:7]: Please note the breaking change on the history.rst  about if the customer is using the Azure CLI version greater than 2.56.1 , these commands will not work and cause breaking change.
For example:

* [Breaking Change]  'az containerapp service': deprecate from Azure CLI version 2.56.1
* 'az containerapp add-on' : support for add-on weaviate create and delete commands

**Comment by bgashirabake** [comment]
@zhoxing-ms is the new change ok?

**Review by Greedygre — APPROVED** [review]

File: src/containerapp/azext_containerapp/commands.py (hunk 1 above)
Comment (zhoxing-ms) [reply src/containerapp/azext_containerapp/commands.py:53]: > Please note that after adding expiration='2.56.1', if the customer is using the Azure CLI version greater than 2.56.1 , these commands will not work and cause breaking change to the customer's automation scripts

@bgashirabake Have you evaluated the impact of this breaking change on customers? If it has almost no impact on customers, then I think this is acceptable

Comment (zhoxing-ms) [reply src/containerapp/azext_containerapp/commands.py:53]: > expiration='2.56.1'

Additionally, we have not yet released version `2.56.1`, the latest CLI version is ``2.56.0`

Comment (bgashirabake) [reply src/containerapp/azext_containerapp/commands.py:53]: @zhoxing-ms yes, we had slated the cut off time for the service command group to be the first release of 2024. Given that i don't know when the 2.56.1 cli version is coming out, i wanted to make it the last version where the service command group is usable. 
About impact: yes, the service group has had a deprecation warning for ~2 months now so we expect customers who use it to be aware. That + the whole time before 2.56.2 gets released seemed reasonable. Moreover, our docs use have used both add-on and service for some time now: https://learn.microsoft.com/en-us/cli/azure/containerapp?view=azure-cli-latest

**Review by zhoxing-ms — APPROVED** [review]

File: src/containerapp/azext_containerapp/commands.py (hunk 1 above)
Comment (Greedygre) [reply src/containerapp/azext_containerapp/commands.py:53]: Hi @bgashirabake @zhoxing-ms 

The version of `2.56.1` is not exist.
The first release of 2024 is `2.56.0`: https://github.com/Azure/azure-cli/milestone/137
The next release is `2.57.0`: https://github.com/Azure/azure-cli/milestone/138

**Comment by bgashirabake** [comment]
@Greedygre @zhoxing-ms, i have a milvus add-on PR that will be out tomorrow. i will make the switch of expiry to 2.57.0, then. thank you all!

**Pull request merged**